### PR TITLE
Issues #53, #54: Remote deployment improvements

### LINKED ISSUE #50: Implicitly call install when starting
Currently, the workflow to install Gigantum from the cli is
```
(1) pip install gigantum
(2) gigantum install
(3) gigantum start
```

It would be great to remove the middle step and have it implicitly install when starting (perhaps with some kind of indication that initial start-up may take some more time).

Ultimately, it would be nice if getting Gigantum up and running is as simple as:

```
pip install gigantum
gigantum start
```

### LINKED ISSUE #53: Add `-y` flag to `gigantum stop` so you don't need an interactive session to shut the Client down
**Description**
In some cases you want to stop gigantum but don't have an interactive session (e.g. a programmatic SSH exec call). This could be fixed by providing a `-y` flag that automatically accepts when running `gigantum stop`.

**Testing Suggestion**
- start gigantum with `gigantum start`
- Should see UI load
- stop gigantum with `gigantum stop -y`
- Should not get prompted to confirm and should see application stop

### LINKED ISSUE #54: Add optional flag to run gigantum in non-default working directory
**Description**
For certain deployments, it is important to be able to set the gigantum working directory to a location other than the default `~/gigantum`. A simple optional flag to `gigantum start` should be added to allow the user to specify the working directory. Updates to the client may be needed to support this.

### PR DESCRIPTION
**Pull request by dkleissa**
The `-y` flag has been added to automatically accept prompts.

The `--working-dir` option has been added to run the client in a non-default working directory.

If the container is not installed on start, you are now prompted to install without having to run an additional command.

Unit tests were added and circleCI configured for this repository.

This PR closes #54, closes #53, and closes #50.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "requirements.txt" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+
+      - run:
+          name: install dependencies
+          command: |
+            python3 -m venv venv
```
Comment (billvb) [inline .circleci/config.yml]: I might suggest like `cli-test-env` rather than venv

File: gigantumcli/cli.py [hunk 2]
```diff
@@ -52,10 +52,21 @@ def main():
...
                         default=60,
                         help="Number of seconds to wait for Client during `start`")
 
+    parser.add_argument("--yes", "-y",
+                        action='store_true',
+                        help="Optional flag to automatically accept confirmation prompts")
+
+    parser.add_argument("--working-dir", "-d",
+                        metavar="<working directory>",
+                        default="~/gigantum",
```
Comment (billvb) [inline gigantumcli/cli.py]: You double check it's properly getting expanded?

File: .circleci/config.yml (hunk 1 above)
Comment (dkleissa) [reply .circleci/config.yml]: this is just boilerplate from circleCI itself. The venv gets created ephemerally in the test, so i think fine. Its sort of the default thing to do when embedding the venv in the repo (e.g. default python .gitignore files will ignore this)

File: gigantumcli/cli.py (hunk 2 above)
Comment (dkleissa) [reply gigantumcli/cli.py]: yes. Gets expanded before use on line 240 of `actions.py`

**Pull request merged**